### PR TITLE
Trash bags tweaks

### PR DESCRIPTION
**Pull request by Macoron**
## About the PR 
Check changelog.

Add new storage fill visualizer. It will receive appearance data from `ServerStorageComponent` and change sprite depending on filled volume. Because security and performance reasons, `ServerStorageComponent` will send volume data only if `showFillLevel` is true.

**Changelog**

:cl:
- tweak: Trash bags now change sprite when you add junk inside.
- tweak: You can dump all trash into disposal from trash bag.

### CODE REVIEW

File: Content.Client/GameObjects/Components/Storage/StorageFillVisualizer.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            if (_maxFillLevels <= 0 || _fillBaseName == null) return;
+
+            if (!component.TryGetData(StorageVisuals.FillLevel,
+                out StorageFillLevel state)) return;
+
+            if (!component.Owner.TryGetComponent(out ISpriteComponent? sprite)) return;
+            if (!sprite.LayerMapTryGet(_layer, out var fillLayer)) return;
+
+            var fillPercent = (float) state.StorageSizeUsed / state.StorageSizeMax;
+            var closestFillSprite = (int) Math.Round(fillPercent * _maxFillLevels);
```
Comment (metalgearsloth) [inline Content.Client/GameObjects/Components/Storage/StorageFillVisualizer.cs:33]: There's 2 ContentHelpers.RoundToLevels but people complain it doesn't round the way they want so ¯\\\_(ツ)\_/¯

File: Content.Server/GameObjects/Components/Disposal/DisposalUnitComponent.cs [hunk 2]
```diff
@@ -615,6 +646,14 @@ void IActivate.Activate(ActivateEventArgs eventArgs)
 
         async Task<bool> IInteractUsing.InteractUsing(InteractUsingEventArgs eventArgs)
         {
+            // you can dump trashbag content 
+            if (eventArgs.Using.HasTag("TrashBag"))
+            {
+                var dumpedTrashbag = TryEmptyTrashbag(eventArgs.User, eventArgs.Using);
+                if (dumpedTrashbag)
+                    return true;
+            }
```
Comment (metalgearsloth) [inline Content.Server/GameObjects/Components/Disposal/DisposalUnitComponent.cs:655]: IMO better off making it a new component and dumping the logic onto a system given it'll make any future changes much easier e.g. if someone wanted to be able to empty one trash bag into another or something.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/GameObjects/Components/Items/Storage/ServerStorageComponent.cs [hunk 3]
```diff
@@ -183,9 +188,20 @@ public void HandleEntityMaybeRemoved(EntRemovedFromContainerMessage message)
...
         }
 
+        private void UpdateFillLevelVisualizer()
+        {
+            // udpate visualizer if needed
+            if (_showFillLevel && _appearanceComponent != null)
+            {
+                var state = new StorageFillLevel(_storageUsed, _storageCapacityMax);
+                _appearanceComponent.SetData(StorageVisuals.FillLevel, state);
+            }
```
[the remark this answers is not in the document]
Comment (Macoron) [reply Content.Server/GameObjects/Components/Items/Storage/ServerStorageComponent.cs:202]: Not sure if I understand. What the difference between `ComponentDependency` and `TryGetComponent`?

[the remark this answers is not in the document]
Comment (metalgearsloth) [reply Content.Server/GameObjects/Components/Items/Storage/ServerStorageComponent.cs:202]: ComponentDependency has the field vs resolving it when needed. Depends how often it gets called as to whether it matters.

**Comment by metalgearsloth** [comment]
You coming back to this Macoron or busy_irl?

**Comment by Macoron** [comment]
Well, I'll better close this for now. If anyone willing to return. feel free to do so.

**Pull request closed without merging**